src/lib/ffi/ffi_cipher.cpp

```diff
@@ -217,7 +244,12 @@ int botan_cipher_get_default_nonce_length(botan_cipher_t cipher, size_t* nl)
 
 int botan_cipher_get_update_granularity(botan_cipher_t cipher, size_t* ug)
    {
-   return BOTAN_FFI_VISIT(cipher, [=](const auto& c) { *ug = c.update_granularity(); });
+   return BOTAN_FFI_VISIT(cipher, [=](const auto& c) { *ug = c.ideal_granularity(); });
```

Comment:
Should this still be `c.update_granularity()`?

Reply:
No because that "idea" granularity is still the required update size on the FFI side of things. If we adopt your suggestion to use the smallest possible multiple from #3163 (which seems like a good idea, but I'll do it in a followup MR) then this would instead return that value.

Reply:
I applied your suggestion from #3163 here and modified this function to return that value instead.